unitig_caller/__main__.py

```diff
@@ -111,95 +120,38 @@ def get_options():
-
-        sys.stderr.write("Building mantis index\n")
-        run_mantis_build(output_list.name, options.output, log_slots=22, mantis_exe=options.mantis)
-        run_mantis_index(options.output, options.cpus, delete_RRR=False, mantis_exe=options.mantis)
-
-        output_list.close()
-
-    elif options.call:
-        sys.stderr.write("Running mantis queries\n")
-
-        unitig_list_file = tempfile.NamedTemporaryFile('w')
-        unitigs = []
-        with open(options.unitigs, 'r') as unitig_file:
-            unitig_file.readline() # header
-            for unitig_line in unitig_file:
-                unitig_fields = unitig_line.rstrip().split("\t")
-                unitigs.append(unitig_fields[0])
-                unitig_list_file.write(unitig_fields[0] + "\n")
-        unitig_list_file.file.flush()
-
-        # Run query and format output as pyseer k-mers/unitigs
-        mantis_index = os.path.join(options.mantis_index, '')
-        with open(options.output + "_unitigs.txt", 'w') as call_output:
-            for unitig, samples in zip(unitigs, run_mantis_query(unitig_list_file.name,
-                                                                 mantis_index,
-                                                                 options.mantis)):
-                if len(samples) > 0:
-                    out_array = [unitig] + ["|"] + [x + ":1" for x in samples]
-                    call_output.write(" ".join(out_array) + "\n")
-
-        unitig_list_file.close()
-
-    elif options.simple:
-        # Read input into lists, as in 'index' and 'call'
-        names_in = []
-        fasta_in = []
-        with open(options.strains, 'r') as strain_file:
-            for strain_line in strain_file:
-                (strain_name, strain_fasta) = strain_line.rstrip().split("\t")
-                names_in.append(strain_name)
-                fasta_in.append(strain_fasta)
-
-        unitigs = []
-        with open(options.unitigs, 'r') as unitig_file:
-            unitig_file.readline() # header
-            for unitig_line in unitig_file:
-                unitig_fields = unitig_line.rstrip().split("\t")
-                unitigs.append(unitig_fields[0])
-
-        # call c++ code to map (also writes output file)
-        map_strings.call(fasta_in,
-                         names_in,
-                         unitigs,
-                         options.output + "_unitigs.txt",
-                         not options.no_save_idx,
-                         options.cpus)
+    if options.build:
+        sys.stderr.write("Building de Bruijn graph with Bifrost\n")
+
+        run_bifrost_build(options.seq, options.output, options.addit_seq, options.no_colour, options.clean, options.kmer_size, options.minimizer_size, options.threads, options.bifrost)
+
+    elif options.query:
+
+        graph_file = options.input + ".gfa"
+        colour_file = options.input+ ".bfg_colors"
+        tsv_file = options.output + ".tsv"
+
+        if options.fasta == None:
```

Comment:
Could you move this into the build section above, so that each time a graph is constructed the unitigs are written out